javascript/ql/lib/semmle/javascript/frameworks/React.qll

```diff
@@ -785,6 +804,15 @@ private class ReactRouterLocationSource extends DOM::LocationSource::Range {
   }
 }
 
+private class UseRefDomValueSource extends DOM::DomValueSource::Range {
+  UseRefDomValueSource() {
+    exists(DataFlow::PropRead current, UseRefStep step | current = this |
+      step.step(_, current) and
+      current.mayHavePropertyName("current")
+    )
```

Comment:
```suggestion
    this =
      any(JsxAttribute attrib | attrib.getName() = "ref")
          .getValue()
          .flow()
          .getALocalSource()
          .getAPropertyRead("current")
```
This is not quite right.

When something is passed as the `ref` attribute in a React component, like `<span ref={foo}`, it means the underlying DOM element is assigned to that ref when it gets mounted. So in this case `foo.current` would become a DOM element reference. I don't think we even have to check that the ref came from `useRef`.

The `useRef()` hook is often used in combination with the `ref={}` attribute, but not always.